### DISCUSSION TITLE
How to supply defaultValues for reset in the child component instead of parent component?

### DISCUSSION THREAD

**Discussion by kevinkli916** [react-hook-form/react-hook-form#1887] [Q&A]
This is a [CSB](https://codesandbox.io/s/react-hook-form-errormessage-gm9d6?file=/src/index.js) for the question.

I want to move defaultValues into SelectComponent.jsx instead of having it in index.js
What is the best way of doing this?

**Comment by bluebill1049** [comment]
why not keep defaultValue inline?

**Reply by kevinkli916** [reply]
The application I am working on is structured like this:

`<form>`
   `<childComponents/>`
`</form>`

The child components are dynamically rendered depending on which view the user is on. The components also contain different user inputs, and to keep track of defaultValues to reset to, I want to put them in the child components if possible.

**Reply by bluebill1049** [reply]
can you produce a simple codesandbox which demonstrate your case above?
